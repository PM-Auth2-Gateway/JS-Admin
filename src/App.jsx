import React from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import styles from './App.module.scss';

import Navigation from './components/Navigation/Navigation';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import ApplicationsPage from './pages/ApplicationsPage/ApplicationsPage';
import SingleApplicationPage from './pages/SingleApplicationPage/SingleApplicationPage';

function App() {
  return (
    <Router>
      <div className={styles.App}>
        <Navigation />
      </div>
      <div className={styles.container}>
        <Switch>
          <PrivateRoute exact path='/applications/:appId' component={SingleApplicationPage} />
          <PrivateRoute exact path='/applications' component={ApplicationsPage} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
