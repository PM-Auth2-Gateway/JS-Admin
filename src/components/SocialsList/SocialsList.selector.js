import { createStructuredSelector } from 'reselect';

const allSocialsSelector = (globalState) => globalState.socials.all.value;
const loadingSocialsSelector = (globalState) => globalState.socials.all.loading;
const errorSocialsSelector = (globalState) => globalState.socials.all.error;

const selector = createStructuredSelector({
  all: allSocialsSelector,
  loading: loadingSocialsSelector,
  error: errorSocialsSelector,
});

export default selector;
