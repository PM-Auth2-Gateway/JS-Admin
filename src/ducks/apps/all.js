import AppsApiService from '../../services/AppsApiService';

const LOAD_ALL = 'apps/all/load';
const ADD_ALL = 'apps/all/add';
const CLEAR_ALL = 'apps/all/clear';

const SET_LOADING = 'apps/all/loading/set';
const SET_ERROR = 'apps/all/error/set';

export const loadAllApps = () => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.get();

    dispatch(loadAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const addApp = ({ name }) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.post({ name });

    dispatch(addAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const clearAllApps = () => (dispatch) => {
  dispatch(clearAll());
};

export const setAppError = (msg) => (dispatch) => {
  dispatch(setError(msg));
};

const loadAll = (apps) => ({
  type: LOAD_ALL,
  payload: apps,
});

const addAll = (app) => ({
  type: ADD_ALL,
  payload: app,
});

const clearAll = () => ({
  type: CLEAR_ALL,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: [],
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD_ALL:
      return {
        ...state,
        value: [...state.value, ...action.payload],
      };
    case ADD_ALL:
      return {
        ...state,
        value: [...state.value, action.payload],
      };
    case CLEAR_ALL:
      return {
        ...state,
        value: [],
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
