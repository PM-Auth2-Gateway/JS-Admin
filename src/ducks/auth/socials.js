import LoginApiService from './../../services/LoginApiService';

const SET_LOADING = 'auth/socials/SET_LOADING';
const SET_SOCIALS = 'auth/socials/SET_SOCIALS';

const setLoading = (payload) => ({
  type: SET_LOADING,
  payload,
});

const setSocials = (payload) => ({
  type: SET_SOCIALS,
  payload,
});

export const getSocials = () => async (dispatch) => {
  dispatch(setLoading(true));
  const socials = await LoginApiService.getSocials();
  dispatch(setSocials(socials));
  dispatch(setLoading(false));
};

const initialState = {
  isLoading: false,
  socials: null,
  authLink: null,
};

const reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case SET_LOADING:
      return {
        ...state,
        isLoading: payload,
      };
    case SET_SOCIALS:
      return {
        ...state,
        socials: payload,
      };
    default:
      return state;
  }
};

export default reducer;
