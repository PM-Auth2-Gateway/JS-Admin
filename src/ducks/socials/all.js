import SocialsApiService from '../../services/SocialsApiService';
import { toast } from 'react-toastify';

const LOAD_ALL = 'socials/all/load';
const UPDATE_STATUS_BY_ID = 'socials/all/update_by_id/status';
const CLEAR_ALL = 'socials/all/clear';

const SET_LOADING = 'socials/all/loading/set';
const SET_ERROR = 'socials/all/error/set';

export const loadAllSocials = (appId) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.get(appId);

    dispatch(loadAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const updateSocialStatusById = (appId, id, { status }) => async (
  dispatch
) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.updateStatusById(appId, id, { status });

    dispatch(
      updateStatusById(id, {
        status: res.data.is_active,
      })
    );
    dispatch(setError(null));

    toast(`Social network was ${!status ? 'disabled' : 'activated'}`);
  } catch (error) {
    dispatch(setError(error));

    toast.error('Something went wrong');
  } finally {
    dispatch(setLoading(false));
  }
};

export const clearAllSocials = () => (dispatch) => {
  dispatch(clearAll());
};

export const setSocialsError = (msg) => (dispatch) => {
  dispatch(setError(msg));
};

const loadAll = (apps) => ({
  type: LOAD_ALL,
  payload: apps,
});

const updateStatusById = (id, { status }) => ({
  type: UPDATE_STATUS_BY_ID,
  payload: {
    id,
    status,
  },
});

const clearAll = () => ({
  type: CLEAR_ALL,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: [],
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD_ALL:
      return {
        ...state,
        value: action.payload,
      };
    case UPDATE_STATUS_BY_ID:
      return {
        ...state,
        value: state.value.map((elem) => ({
          ...elem,
          is_active:
            elem.id === action.payload.id
              ? action.payload.status
              : elem.is_active,
        })),
      };
    case CLEAR_ALL:
      return {
        ...state,
        value: [],
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
