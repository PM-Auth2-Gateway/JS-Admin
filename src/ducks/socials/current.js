import SocialsApiService from '../../services/SocialsApiService';

const SET_CURRENT = 'socials/current/set';
const UPDATE_CURRENT = 'socials/current/update';
const REMOVE_CURRENT = 'socials/current/remove';

const SET_LOADING = 'socials/current/loading/set';
const SET_ERROR = 'socials/current/error/set';

export const loadCurrentSocial = (appId, id) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.getById(appId, id);

    dispatch(setCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const updateCurrentSocial = (
  appId,
  id,
  { client_id, scope, secret_key }
) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.updateById(appId, id, {
      client_id,
      scope,
      secret_key,
    });

    dispatch(updateCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const removeCurrentSocial = () => (dispatch) => {
  dispatch(removeCurrent());
};

const setCurrent = (app) => ({
  type: SET_CURRENT,
  payload: app,
});

const updateCurrent = (data) => ({
  type: UPDATE_CURRENT,
  payload: data,
});

const removeCurrent = () => ({
  type: REMOVE_CURRENT,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: {},
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case SET_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case UPDATE_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case REMOVE_CURRENT:
      return {
        ...state,
        value: {},
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
