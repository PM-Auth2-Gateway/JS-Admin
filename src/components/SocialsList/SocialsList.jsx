import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useAppContext } from '../../contexts/App.context';
import { Spinner } from 'react-bootstrap';

import SocialPreview from '../SocialPreview/SocialPreview';

import { loadAllSocials, clearAllSocials } from '../../ducks/socials/all';
import selector from './SocialsList.selector';

const SocialsList = () => {
  const dispatch = useDispatch();

  const { appId } = useAppContext();
  const { all, loading } = useSelector(selector);

  useEffect(() => {
    dispatch(loadAllSocials(appId));

    return () => {
      dispatch(clearAllSocials());
    };
  }, [dispatch, appId]);

  return loading ? (
    <Spinner animation='border' />
  ) : (
    <>
      {all.map(({ id, name, logo_path, is_setting, is_active }) => (
        <SocialPreview
          key={id}
          id={id}
          name={name}
          is_setting={is_setting}
          is_active={is_active}
          logo_path={logo_path}
        />
      ))}
    </>
  );
};

export default SocialsList;
