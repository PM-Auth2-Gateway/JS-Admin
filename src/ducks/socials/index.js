import { combineReducers } from 'redux';

import { reducer as socialsAllReducer } from './all';
import { reducer as socialsCurrentReducer } from './current';

export const reducer = combineReducers({
  all: socialsAllReducer,
  current: socialsCurrentReducer,
});
