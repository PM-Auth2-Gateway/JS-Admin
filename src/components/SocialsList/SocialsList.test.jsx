import { shallow } from 'enzyme';
import { useDispatch, useSelector } from 'react-redux';
import { useAppContext } from '../../contexts/App.context';

import { loadAllSocials, clearAllSocials } from '../../ducks/socials/all';

import SocialsList from './SocialsList';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('../../contexts/App.context', () => ({
  useAppContext: jest.fn(),
}));

jest.mock('../../ducks/socials/all', () => ({
  loadAllSocials: jest.fn(),
  clearAllSocials: jest.fn(),
}));

describe('SocialsList Component', () => {
  const dispatch = jest.fn();
  const action = jest.fn();

  beforeAll(() => {
    loadAllSocials.mockReturnValue(action);
    clearAllSocials.mockReturnValue(action);
  });

  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);

    useAppContext.mockReturnValue({
      appId: 1,
    });
    useSelector.mockReturnValue({
      all: [
        {
          id: 1,
          name: 'Google',
          isSetting: false,
        },
        {
          id: 1,
          name: 'Facebook',
          isSetting: false,
        },
      ],
      loading: false,
      error: null,
    });
  });

  it('should render component', function () {
    const component = shallow(<SocialsList />);

    expect(component).toMatchSnapshot();
  });
});
