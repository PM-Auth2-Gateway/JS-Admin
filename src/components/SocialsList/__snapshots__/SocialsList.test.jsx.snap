// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialsList Component should render component 1`] = `
<Fragment>
  <SocialPreview
    id={1}
    key="1"
    name="Google"
  />
  <SocialPreview
    id={1}
    key="1"
    name="Facebook"
  />
</Fragment>
`;
